### PR TITLE
Add Azurite Node devcontainer

### PR DESCRIPTION
**Pull request by AlexanderYW**
This is a new devcontainer that setup recommended extensions and ports for running Azurite locally in a devcontainer

It also contains two tests for `blob` and `queue`

### CODE REVIEW

File: containers/javascript-node-azurite/.devcontainer/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    # Runs app on the same network as the database container, allows "forwardPorts" in devcontainer.json function.
+    network_mode: service:storage
+
+    # Uncomment the next line to use a non-root user for all processes.
+    # user: node
+
+    # Use "forwardPorts" in **devcontainer.json** to forward an app port locally. 
+    # (Adding the "ports" property to this file will not forward from a Codespace.)
+
+  storage:
```
Comment (ItalyPaleAle) [inline containers/javascript-node-azurite/.devcontainer/docker-compose.yml:30]: Would it make sense to add `container_name: storage` (or `azurite`) so the other services can connect to Azurite via `http://storage:10000` directly?

File: containers/javascript-node-azurite/.devcontainer/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    image: mcr.microsoft.com/azure-storage/azurite:latest
+    restart: unless-stopped
+    volumes:
+      - storage-data:/data/db
```
Comment (ItalyPaleAle) [inline containers/javascript-node-azurite/.devcontainer/docker-compose.yml:34]: Should the entire `/data` folder be mounted in the container?  As per docs: https://github.com/Azure/Azurite#run-azurite-v3-docker-image-with-customized-persisted-data-location

**Review by ItalyPaleAle — COMMENTED** [review]
I've used Azurite in a dev container before and I think this is a good example, but I've proposed a few tweaks (I like how this should work through Codespaces too!)

File: containers/javascript-node-azurite/.devcontainer/docker-compose.yml (hunk 2 above)
Comment (AlexanderYW) [reply containers/javascript-node-azurite/.devcontainer/docker-compose.yml:34]: Thanks for pointing that out, I will correct that soon :)

File: containers/javascript-node-azurite/.devcontainer/docker-compose.yml (hunk 1 above)
Comment (AlexanderYW) [reply containers/javascript-node-azurite/.devcontainer/docker-compose.yml:30]: Sure, I can add that :)

**Comment by AlexanderYW** [comment]
> I've used Azurite in a dev container before and I think this is a good example, but I've proposed a few tweaks (I like how this should work through Codespaces too!)

To be honest with you, I tried reaching out to a few friends if they had gotten early access to codespaces, so they could test it for me since I haven't, but no one had, if you have, I would love if you could test it out, but based on my assumptions it should 🙃

File: containers/javascript-node-azurite/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+# Node.js & Azurite
```
Comment (2percentsilk) [inline containers/javascript-node-azurite/README.md:1]:
```suggestion
# Node.js & Azurite (Community)
```

File: containers/javascript-node-azurite/README.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+## Summary
+
+Everything you need to get started using Node with Azurite.
+
+| Metadata | Value |  
+|----------|-------|
+| *Contributors* | [Alexander Wennerstrøm](https://github.com/AlexanderYW)   |
+| *Definition type* | Docker Compose |
```
Comment (2percentsilk) [inline containers/javascript-node-azurite/README.md:10]: We've added a new piece of metadata: Categories! 

```suggestion
| *Categories* | Community, Azure
| *Definition type* | Docker Compose |
```

**Review by 2percentsilk — CHANGES_REQUESTED** [review]
Apologies for the delay here. This addition looks great. I suggested two small changes since we've updated how we title and tag the definitions so you'll need to please accept those.

As the final piece, I want to confirm that you're willing to be the maintainer for this definition - generally this means we'll loop you in for comment/approval for any other PRs that the community may make against this definition. Once you confirm this, I'll be happy to merge!

**Comment by AlexanderYW** [comment]
Sure, I can confirm that I'm willing to maintain this 🙂

**Pull request merged**